tests/tests_lite/test_lite.py

```diff
@@ -541,3 +542,18 @@ def run(self):
         TypeError, match=escape("Overriding `LightningLite.run()` and launching from the CLI is not allowed")
     ):
         LiteWithRun()
+
+
+def test_module_sharding_context():
```

Comment:
if we decide to make `_Sharded` a protocol, can we add some tests that our sharded strategies actually pass the instance tests of this protocol (once added and use a dummy class in the meantime)?